bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -611,26 +612,10 @@ public boolean isAborted() {
 
   @Override
   public void truncateTable(TableName tableName, boolean preserveSplits) throws IOException {
-    byte[][] splitKeys = null;
-    if (preserveSplits) {
-      splitKeys = getSplits(tableName);
-    }
-    HTableDescriptor tableDescriptor = getTableDescriptor(tableName);
-    LOG.info("Deleting table %s for truncation.", tableName);
-    deleteTable(tableName);
-    LOG.info("Recreating table %s for truncation", tableName);
-    createTable(tableDescriptor, splitKeys);
-  }
-
-  private byte[][] getSplits(TableName tableName) throws IOException {
-    try (RegionLocator regionLocator = connection.getRegionLocator(tableName)) {
-      List<HRegionLocation> allRegionLocations = regionLocator.getAllRegionLocations();
-      byte[][] splits = new byte[allRegionLocations.size() - 1][];
-      for (int i = 0; i < allRegionLocations.size() - 1; i++) {
-        splits[i] = allRegionLocations.get(i).getRegionInfo().getEndKey();
-      }
-      return splits;
-    }
+    bigtableTableAdminClient.deleteRowRange(BulkDeleteRowsRequest.newBuilder()
+        .setTableName(options.getClusterName().toTableNameStr(tableName.getNameAsString()))
+        .setDeleteAllDataFromTable(true)
+        .build());
```

Comment:
} catch (Throwable throwable) {
    throw new IOException(
            String.format(
                    "Failed to truncate table '%s'",
                    tableName.getNameAsString()),
            throwable);
  }